CMakeLists.txt

```diff
@@ -1,54 +1,105 @@
 cmake_minimum_required(VERSION 3.5)
-set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_C_STANDARD 99)
+set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+set(CMAKE_POLICY_DEFAULT_CMP0063 NEW)
 project(Fossilize LANGUAGES CXX C)
 
-option(FOSSILIZE_MSVC_ANALYZE "Enable /analyze" OFF)
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+set(CMAKE_THREAD_PREFER_PTHREAD ON)
+
+include(CMakeDependentOption)
+include(GNUInstallDirs)
+include(CTest)
+
+find_package(Threads REQUIRED)
+
+if (NOT DEFINED CMAKE_CXX_STANDARD OR CMAKE_CXX_STANDARD LESS 14)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if (NOT DEFINED CMAKE_C_STANDARD OR CMAKE_C_STANDARD LESS 99)
+  set(CMAKE_C_STANDARD 99)
+endif()
+
+if (NOT DEFINED CMAKE_CXX_VISIBILITY_PRESET)
+  set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+endif()
+
+if (NOT DEFINED CMAKE_C_VISIBILITY_PRESET)
+  set(CMAKE_C_VISIBILITY_PRESET hidden)
+endif()
+
+cmake_dependent_option(FOSSILIZE_TESTS
+  "Enable Fossilize tests" ON
+  "CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME;BUILD_TESTING" OFF)
+
 option(FOSSILIZE_ENABLE_EXCEPTIONS "Compile Fossilize with exception support." OFF)
+option(FOSSILIZE_MSVC_ANALYZE "Enable /analyze" OFF)
+# Not needed for position-independent-code, as its always turned on
+option(FOSSILIZE_VULKAN_LAYER "Build Vulkan layer." ON) 
+option(FOSSILIZE_CLI "Enable Fossilize CLI support." ON)
+
+option(FOSSILIZE_SANITIZE_ADDRESS "Sanitize address" OFF)
+option(FOSSILIZE_SANITIZE_THREADS "Sanitize threads" OFF)
 
-if (CMAKE_COMPILER_IS_GNUCXX OR (${CMAKE_CXX_COMPILER_ID} MATCHES "Clang"))
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fvisibility=hidden")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden")
-    if (NOT FOSSILIZE_ENABLE_EXCEPTIONS)
-        set(FOSSILIZE_EXCEPTION_FLAGS "-fno-exceptions")
-    else()
-        set(FOSSILIZE_EXCEPTION_FLAGS "")
-    endif()
-    set(FOSSILIZE_CXX_FLAGS -Wall -Wextra -Wunused-result -Wno-missing-field-initializers -Wno-empty-body -Wshadow ${FOSSILIZE_EXCEPTION_FLAGS})
+# option() cannot technically be used in newer CMake versions for non-boolean values
+# warnings will be thrown and this can be annoying otherwise.
+set(FOSSILIZE_VULKAN_INCLUDE_PATH "${PROJECT_SOURCE_DIR}/khronos/vulkan"
```

Comment:
Can this still be overriden by API users? I do that in another project.

Reply:
Yes. If a project "above" Fossilize (i.e., fossilize is used via `add_subdirectory`) defines this variable, then this call will have no effect. Likewise, if you use `-DFOSSILIZE_VULKAN_INCLUDE_PATH=<path/to/vulkan>` when configuring CMake, this call will also have no effect :)